### PR TITLE
Fix for Remorph of a call with nonstandard args

### PR DESCRIPTION
**Pull request by briansull**
On ARM64 when we have a fixed return buffer arg and also have nonstandard args
We will hit an assert when we re-morph the call after an optimization such as a null check elimination
This change corrects this issue

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
I merged this job, I discussed this change set with Bruce.
This change unblocks Rahul pending change set.
I plan to prototype changing the Arm64 return buffer into a non standard arg which should simplify the processing of it here.

File: src/jit/morph.cpp [hunk 1]
```diff
@@ -3739,21 +3755,19 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
...
 #endif // defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
-                    );
-                (void)newArg; //prevent "unused variable" error from GCC
-
-                newArg->SetIsHfaRegArg(passUsingFloatRegs && isHfaArg); // Note on Arm32 a HFA is passed in int regs for varargs
+                                                                    );
+                argEntry->SetIsHfaRegArg(passUsingFloatRegs && isHfaArg); // Note on Arm32 a HFA is passed in int regs for varargs
 
 #ifdef _TARGET_ARM_
                 newArg->SetIsBackFilled(isBackFilled);
```
Comment (prajwal-aithal) [inline src/jit/morph.cpp]: @briansull Shouldn't newArg be changed to argEntry here as well? I see a
> error: use of undeclared identifier 'newArg'
>      newArg->SetIsBackFilled(isBackFilled);
>      ^

error on ARM builds.

**Comment by prajwal-aithal** [comment]
@briansull Change in the variable name (from `newArg` to `argEntry`) in lines `3758`~`3776` is breaking ARM builds (due to `newArg` being used in line `3773`).

Could you please confirm if the variable name should indeed be changed in line `3773`?

**Comment by lemmaa** [comment]
#5914

File: src/jit/morph.cpp (hunk 1 above)
Comment (briansull) [inline src/jit/morph.cpp]: Yes

**Pull request merged**